aspnetcore/fundamentals/minimal-apis/bindStreamPipeReader/7.0-samples/PipeStreamToBackgroundQueue/Program.cs

```diff
@@ -0,0 +1,47 @@
+using System.Threading.Channels;
+using BackGroundQueueService;
+var builder = WebApplication.CreateBuilder(args);
+// Create a channel to send data to the background queue.
+builder.Services.AddSingleton<Channel<Stream>>((_) => Channel.CreateUnbounded<Stream>());
+
+// Create a background queue service.
+builder.Services.AddHostedService<BackGroundQueue>();
+var app = builder.Build();
+
+app.Use(async (context, next) =>
+{
+    await next();
+    var reader = new StreamReader(context.Request.Body);
+    app.Logger.LogInformation("Reading the stream from the request body.");
+});
+
+app.MapGet("/", () => "Hello World!");
+
+// curl --request POST 'http://localhost:5256/register' --header 'Content-Type: application/json' --data-raw '{ "Name":"Samson", "Age": 23, "Country":"Nigeria" }'
+app.MapPost("/register", async (Stream body, HttpRequest req, Channel<Stream> queue) =>
```

Comment:
This API should return a 202 accepted. I'm not sure why the stream is being reused.

Reply:
for logging and also for using the stream in a queue

Reply:
> This API should return a 202 accepted. I'm not sure why the stream is being reused.

Ok would change that